### PR TITLE
[aws-cloudwatch-metrics] Introduce configuration to support statsd metrics

### PR DESCRIPTION
**Pull request by ltamm**
### Issue

This PR introduces configuration that allows the Cloudwatch agent daemonset to collect statsd metrics. 

Pulled in commits from an open PR: https://github.com/aws/eks-charts/pull/514
Related issue: https://github.com/aws/eks-charts/issues/413

### Description of changes

* Pulled in changes from pre-existing unmerged PR which provided the basic configuration: https://github.com/aws/eks-charts/pull/514
* Added values to optionally override some default settings in the configuration file (cloudwatch namespace, metrics collection and aggregation intervals)
* Updated README with new configuration values

### Checklist
- [x] Added/modified documentation as required (such as the `README.md` for modified charts)
- [x] Incremented the chart `version` in `Chart.yaml` for the modified chart(s)
- [x] Manually tested. Describe what testing was done in the testing section below
- [x] Make sure the title of the PR is a good description that can go into the release notes

### Testing

* Ran `helm lint`
* Deployed successfully to our test cluster with custom namespace specified, and observed metrics being sent to cloudwatch
* Ran `helm template` with and without the new options and manually validated the config map contents. Example:
```
$ # Statsd not enabled
$ helm template stable/aws-cloudwatch-metrics | yq 'select (.kind == "ConfigMap") | .data'                                                                                  
cwagentconfig.json: |
  {
    "logs": {
      "metrics_collected": {
        "kubernetes": {
          "cluster_name": "cluster_name",
          "metrics_collection_interval": 60
        }
      },
      "force_flush_interval": 5
    }
  }

$ # Statsd enabled
$ helm template stable/aws-cloudwatch-metrics --set statsd.enabled=true | yq 'select (.kind == "ConfigMap") | .data'                                                                               
cwagentconfig.json: |
  {
    "logs": {
      "metrics_collected": {
        "kubernetes": {
          "cluster_name": "cluster_name",
          "metrics_collection_interval": 60
        }
      },
      "force_flush_interval": 5
    },
    "metrics": {
      "metrics_collected": {
        "statsd": {
            "service_address": ":8125"
        }
      }
    }
  }

$ # Statsd enabled with custom overrides
$ helm template stable/aws-cloudwatch-metrics \
	--set statsd.enabled=true \
	--set statsd.cloudwatch_namespace="custom_namespace" \
	--set statsd.metrics_aggregation_interval=10 \
	--set statsd.metrics_collection_interval=15 \
	--set statsd.port=8135 \
	| yq 'select (.kind == "ConfigMap") | .data'                                                                          
cwagentconfig.json: |
  {
    "logs": {
      "metrics_collected": {
        "kubernetes": {
          "cluster_name": "cluster_name",
          "metrics_collection_interval": 60
        }
      },
      "force_flush_interval": 5
    },
    "metrics": {
      "namespace": "custom_namespace",
      "metrics_collected": {
        "statsd": {
            "metrics_aggregation_interval": 10,
            "metrics_collection_interval": 15,
            "service_address": ":8135"
        }
      }
    }
  }
```
* Validated the port and protocol are reflected successfully in the daemonset:
```
$ helm template stable/aws-cloudwatch-metrics --set statsd.enabled=true | yq 'select (.kind == "DaemonSet") | .spec.template.spec.containers[0].ports'
- containerPort: 8125
  hostPort: 8125
  protocol: UDP
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by kishorj — COMMENTED** [review]
I have some nit comments, changes look good otherwise

File: stable/aws-cloudwatch-metrics/templates/daemonset.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ spec:
       - name: {{ .Chart.Name }}
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- include "aws-cloudwatch-metrics.statsdConfig" . | nindent 8 }}
```
[the remark this answers is not in the document]
Comment (ltamm) [reply stable/aws-cloudwatch-metrics/templates/daemonset.yaml:22]: I can't replicate this. I tried rendering without any sets and it looked right to me
`helm template stable/aws-cloudwatch-metrics` outputs this (truncated):
```
      containers:
        - name: aws-cloudwatch-metrics
          image: "amazon/cloudwatch-agent:1.247350.0b251780"
          imagePullPolicy: IfNotPresent
          # Please don't change below envs
          env:
```

There is a conditional check in the referenced helper function: https://github.com/ltamm/eks-charts/blob/master/stable/aws-cloudwatch-metrics/templates/_helpers.tpl#L64

Apologies if I misunderstood the concern here!

[the remark this answers is not in the document]
Comment (kishorj) [reply stable/aws-cloudwatch-metrics/templates/daemonset.yaml:22]: This is what I see:
helm template . 
```
      containers:
      - name: aws-cloudwatch-metrics
        image: "amazon/cloudwatch-agent:1.247350.0b251780"
        imagePullPolicy: IfNotPresent
        
        # Please don't change below envs
```
There is newline in the manifest. I'm using helm version v3.11.2

helm template .  --set statsd.enabled=true
```
        image: "amazon/cloudwatch-agent:1.247350.0b251780"
        imagePullPolicy: IfNotPresent
        ports:
          - containerPort: 8125
            hostPort: 8125
            protocol: UDP
        # Please don't change below envs
```
This one is ok.

[the remark this answers is not in the document]
Comment (ltamm) [reply stable/aws-cloudwatch-metrics/templates/daemonset.yaml]: aha, I upgraded helm and got the same output. I think it's fixed now:

helm template .
```
        image: "amazon/cloudwatch-agent:1.247350.0b251780"
        imagePullPolicy: IfNotPresent
        # Please don't change below envs
```

helm template . --set statsd.enabled=true
```
        image: "amazon/cloudwatch-agent:1.247350.0b251780"
        imagePullPolicy: IfNotPresent
        ports:
          - containerPort: 8125
            hostPort: 8125
            protocol: UDP
        # Please don't change below envs
```

**Review by kishorj — APPROVED** [review]

**Pull request merged**